backend/rest/rest.go

```diff
@@ -35,53 +38,56 @@ func New(token string, resultsPerPage int, storageProvider persistence.Storage)
 	return &client{githubClient, ctx, resultsPerPage}, nil
 }
 
-// Search searches code using the github api
-func (c *client) Search(query string, lang string) map[int64]*backends.Result {
+var repos map[int64]*backends.Result
 
+// Search searches code using the github api
+func (c *client) Search(query, lang string) map[int64]*backends.Result {
 	opt := &github.SearchOptions{
 		ListOptions: github.ListOptions{PerPage: c.resultsPerPage},
 	}
 
 	results, _, err := c.Client.Search.Code(c.ctx, query, opt)
-	repos := make(map[int64]*backends.Result)
-
 	if err != nil {
 		log.Fatal("Failed to fetch results", err)
 	}
 
+	var repoIDs []int64
 	for _, codeResult := range results.CodeResults {
-		repoID := codeResult.GetRepository().GetID()
-		fmt.Printf("Fetching value for: %v\n", repoID)
-		value, err := storage.Get(repoID)
-		fmt.Printf("Found value: %s for repoID: %v\n", value, repoID)
+		repoIDs = append(repoIDs, codeResult.GetRepository().GetID())
+	}
 
-		if err != nil || value == nil {
-			repo, _, err := c.Client.Repositories.GetByID(c.ctx, repoID)
+	values, err := storage.Get(repoIDs)
+	if err != nil {
+		fmt.Println(err)
+	}
 
+	cachedRepoIds := initializeCache(values)
```

Comment:
So the entire cache is to cache the calls to GitHub - what about memory cache to sqlite? should we even do it?

Reply:
this is the memory cache, no?
I don't know if something more complex is required given this is a short-living program